source/optional.d.ts

```diff
@@ -0,0 +1,14 @@
+/**
+Makes a type optional. Allows a type to be `undefined`, but not `null`.
+
```

Comment:
> Write about some real-world use-cases where it can be useful. (It can be hard sometimes for users to see where they would use something) - https://github.com/sindresorhus/type-fest/blob/master/.github/contributing.md